packages/backend/src/models/ProjectModel/ProjectModel.ts

```diff
@@ -1802,4 +1806,60 @@ export class ProjectModel {
     getWarehouseClientFromCredentials(credentials: CreateWarehouseCredentials) {
         return warehouseClientFromCredentials(credentials);
     }
+
+    async createCustomExplore(
+        projectUuid: string,
+        { name, sql, columns }: CreateCustomExplorePayload,
+        warehouseClient: WarehouseClient,
+    ): Promise<Pick<Explore, 'name'>> {
+        const translatedToExplore = createCustomExplore(
+            name,
+            sql,
+            columns,
+            warehouseClient,
+        );
+
+        // insert into cached_explore
+        await this.database(CachedExploreTableName)
+            .insert({
+                project_uuid: projectUuid,
+                name: translatedToExplore.name,
+                table_names: Object.keys(translatedToExplore.tables || {}),
+                explore: JSON.stringify(translatedToExplore),
+            })
+            .onConflict(['project_uuid', 'name'])
+            .merge()
+            .returning(['name', 'cached_explore_uuid']);
+        const toAddToCached: Explore = {
+            name,
+            tags: [],
+            label: name,
+            tables: translatedToExplore.tables,
+            baseTable: name,
+            groupLabel: name,
```

Comment:
no need to add a gropuLabel, or perhaps we could use `sql runner` instead, so it doesn't look like this on the explore: 

![image](https://github.com/user-attachments/assets/04d33377-9091-4427-a605-cee70ba7482c)

Reply:
fixed the `groupLabel` and made some tiny UI/UX improvements with @PriPatel 👍